modules/sharedaddy/sharing.js

```diff
@@ -1,364 +1,474 @@
-					facebook: [
-						window.location.protocol +
-							'//graph.facebook.com/?callback=WPCOMSharing.update_facebook_count&ids=' +
-							encodeURIComponent( url ),
-					],
-				};
-
-				for ( service in requests ) {
-					if ( ! jQuery( 'a[data-shared=sharing-' + service + '-' + id + ']' ).length ) {
-						continue;
-					}
+	// -------------------------- UTILITY FUNCTIONS -------------------------- //
 
-					while ( ( service_request = requests[ service ].pop() ) ) {
-						jQuery.getScript( service_request );
-					}
+	// Helper function to load an external script.
+	function loadScript( url ) {
+		var script = document.createElement( 'script' );
+		var prev = currentScript || document.getElementsByTagName( 'script' )[ 0 ];
+		script.async = true;
+		script.src = url;
+		prev.parentNode.insertBefore( script, prev );
+	}
 
-					if ( sharing_js_options.is_stats_active ) {
-						WPCOMSharing.bump_sharing_count_stat( service );
-					}
-				}
+	// Helper closest parent node function (not a polyfill) based on
+	// https://developer.mozilla.org/en-US/docs/Web/API/Element/closest#Polyfill
+	function closest( el, sel ) {
+		if ( el.closest ) {
+			return el.closest( sel );
+		}
 
-				WPCOMSharing.done_urls[ id ] = true;
-			}
-		},
+		var matcher =
+			Element.prototype.matches ||
+			Element.prototype.msMatchesSelector ||
+			Element.prototype.webkitMatchesSelector;
+		var current = el;
 
-		// get the version of the url that was stored in the dom (sharing-$service-URL)
-		get_permalink: function ( url ) {
-			if ( 'https:' === window.location.protocol ) {
-				url = url.replace( /^http:\/\//i, 'https://' );
-			} else {
-				url = url.replace( /^https:\/\//i, 'http://' );
+		do {
+			var matches = matcher.bind( current );
+			if ( matches( sel ) ) {
+				return current;
 			}
+			current = current.parentElement || current.parentNode;
+		} while ( current !== null && current.nodeType === 1 );
 
-			return url;
-		},
-		update_facebook_count: function ( data ) {
-			var url, permalink;
+		return null;
+	}
 
-			if ( ! data ) {
-				return;
-			}
+	// Helper function to iterate over a NodeList
+	// (since IE 11 doesn't have NodeList.prototype.forEach)
+	function forEachNode( list, fn ) {
+		for ( var i = 0; i < list.length; i++ ) {
+			var node = list[ i ];
+			fn( node, i, list );
+		}
+	}
 
-			for ( url in data ) {
-				if (
-					! data.hasOwnProperty( url ) ||
-					! data[ url ].share ||
-					! data[ url ].share.share_count
-				) {
-					continue;
-				}
+	// Helper function to remove a node from the DOM.
+	function removeNode( node ) {
+		if ( node && node.parentNode ) {
+			node.parentNode.removeChild( node );
+		}
+	}
 
-				permalink = WPCOMSharing.get_permalink( url );
+	// Helper functions to show/hide a node, and check its status.
+	function hideNode( node ) {
+		if ( node ) {
+			node.style.display = 'none';
+		}
+	}
 
-				if ( ! ( permalink in WPCOM_sharing_counts ) ) {
-					continue;
-				}
+	function showNode( node ) {
```

Comment:
This also doesn't need to be a function when it's a one-liner.

Reply:
Same as above.